docs/fee_design.md

```diff
@@ -0,0 +1,173 @@
+1. There can be at most one fee transaction type per block. (PS. as we extends the Plasma Framework, there can potentially be multiple transaction types that all counts as "fee tx type")
+
+Since the transactions are automatically generated from the child chain service, there is no need to do an authentication check. Inclusion of fee transaction is optional, however fees collected in omitted blocks are lost to claim.
+
+### Transaction design
+The transaction would be of the following format:
+
+```
+        {
+                txType: xxx,
+                inputs: [],
+                outputs: [],
+                nonce: xxx,
+        }
+```
+
+The input of the fee transaction can just be empty.
+
+It would have one output per transaction, which follows fungible token output format, which can be spent to a Payment transaction.
+
+```
+        {
+                outputType: xxx,
+                amount: xxx,
+                outputGuard: xxx,
+                token: xxx,
+        }
+```
+For simplicity, a fee transaction would have just one output. To represent fee for multiple tokens, one would need to generate multiple fee transactions.
+
+The last field `nonce` would be `hash(blockNum, token)`. `blockNum` is the block number that the fee tx is mined to. `token` is the token that is claimed in the fee tx output. This combination promises the `nonce` to be unique and thus promises the fee tx to be unique.
+
+Since this fee transaction would have a specific transaction type, (and also it's outputs would have unique output types) we don’t need to worry about different transaction type that uses the same mechanism (block number + token) to ensure uniqueness. Even if another transaction type, for instance, collides with the same outputs in same block, they would end up with different transaction hash due to the transaction's type specifier difference. Transaction uniqueness is granted.
+
+### Fee transaction type extension
+Current Plasma Framework design is immutable on the ability of spending an output type in another tx type once contract deployed. So let's say we have payment v1 that can spend the fee claiming output. However, when we extend the framework with payment v2, we would need another fee output type that is able to be spent in payment v2. As a result, we would need fee tx type 2 as well during the extension. (As only new transaction type can create new output type)
+
+Given this, the block verification should limit the block to be existing with a singe fee tx type per block for simplicity. No matter which fee tx types (1 or 2), it should calculate the fee balance of all transaction within the block. The only difference is how the output can be spent.
+
+Child chain and watcher could even further deprecate old fee tx types afterward if not necessary anymore. The logic change could be done by upgrading child chain service and watcher together.
+
+### Fee transaction type extension
```

Comment:
Oh hey - duplicated chapter see :point_up:

Reply:
Yep removed it.